packages/cli/src/oclif/commands/invoke.js

```diff
@@ -300,7 +312,261 @@ const customLogger = (message, data) => {
   debug(data);
 };
 
+const formatFieldDisplay = (field) => {
+  const ftype = field.type || 'string';
+  let result;
+  if (field.label) {
+    result = `${field.label} | ${field.key} | ${ftype}`;
+  } else {
+    result = `${field.key} | ${ftype}`;
+  }
+  if (field.required) {
+    result += ' | required';
+  }
+  return result;
+};
+
 class InvokeCommand extends BaseCommand {
+  async promptForAuthFields(authFields) {
+    const authData = {};
+    for (const field of authFields) {
+      if (field.computed) {
+        continue;
+      }
+      const message = formatFieldDisplay(field) + ':';
+      let value;
+      if (field.type === 'password') {
+        value = await this.promptHidden(message, true);
+      } else {
+        value = await this.prompt(message, { useStderr: true });
+      }
+      authData[field.key] = value;
+    }
+    return authData;
+  }
+
+  async startBasicAuth(authFields) {
+    if (this.nonInteractive) {
```

Comment:
Basically we want to make sure the CLI doesn't ask interactive questions in non-interactive mode (`nonInteractive === true`). For `basic` auth, the only thing that `auth start` does is ask for username and password interactively. `auth start` with `basic` auth doesn't really make sense in non-interactive mode, which is why we throw an error here.

Does that make sense?

Reply:
In case you're asking why we're not allowing this for `basic` auth:

```
echo '{"username":"user","password":"secret"}' | zapier invoke auth start
```

We could but that I thought that would be useless, since the user can just edit `.env` or append the auth fields to `.env`:

```
echo 'authData_username=user' >> .env
echo 'authData_password=secret' >> .env
```